.github/workflows/check-merge-freeze.yml

```
@@ -0,0 +1,23 @@
+name: Merge freeze
+
+on:
+  pull_request:
+    types: [opened, ready_for_review, synchronize, reopened, labeled, unlabeled]
+    branches: [main]
+  merge_group:
+    types: [checks_requested]
+
+jobs:
+  check-merge-freeze:
+    name: Check
+    # This condition is to avoid blocking the PR causing the freeze in the first place.
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'release:merge-freeze') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Can we use the version we use elsewhere? https://github.com/open-telemetry/opentelemetry-collector/blob/e27c4aeae8da140e286534c69fb9f8231b4b79a2/.github/workflows/changelog.yml#L29

To follow [best practices](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) we refer to the version by its full length commit SHA with a comment specifying the version (tooling like dependabot [support this natively](https://github.blog/changelog/2022-10-31-dependabot-now-updates-comments-in-github-actions-workflows-referencing-action-versions/)):

```suggestion
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```